### PR TITLE
add customization `kokkosinlinefunction` to sphinx 

### PR DESCRIPTION
**Pull request by antoinemeyer5**
This PR adds the following customization to the `cppkokkos` domain to automatically add markup `KOKKOS_INLINE_FUNCTION`:

# CODE example 

```
``Title``
=========

.. role::cpp(code)
    :language: cpp

.. role:: cppkokkos(code)
    :language: cppkokkos

Section
-------

.. cpp:class:: MyCoolClass

    .. cpp:function:: void nonTemplatedFunc(double d) ;

    .. cppkokkos:kokkosinlinefunction:: nonTemplatedFunc(double arg) ;

    .. cpp:function:: template<class T> void templatedFunc(const T& arg) ;

    .. cppkokkos:kokkosinlinefunction:: template<class T> void templatedFunc(const T & arg) ;
```

# Result

image

### CODE REVIEW

File: docs/source/_ext/cppkokkos.py [hunk 1]
```diff
@@ -2866,28 +2866,28 @@ def get_id(self, version: int, objectType: str = None,
...
+                          res.append(typeId)
+                          res.append(returnTypeId)
+                  res.append(self.decl.get_param_id(version))
+             elif objectType == 'type':  # just the name
+                 res.append(symbol.get_full_nested_name().get_id(version))
+             elif objectType == 'deprecated-type':  # just the name
+                 res.append(symbol.get_full_nested_name().get_id(version))
+             else:
+                 print(objectType)
+                 assert False
```
Comment (fnrizzi) [inline docs/source/_ext/cppkokkos.py:2890]: @antoinemeyer5 can you please fix this indentation? I think it is my fault but can you please make indentation to be 4 characters?

Comment (fnrizzi) [reply docs/source/_ext/cppkokkos.py:2890]: those weird green things make me think the indentation is set to 5

**Review by fnrizzi — CHANGES_REQUESTED** [review]
see my comment about indentation

**Review by fnrizzi — APPROVED** [review]

**Review by JBludau — APPROVED** [review]
looks good

**Review by cz4rs — APPROVED** [review]

**Comment by fnrizzi** [comment]
the name was suggested/approved during team meeting on Jan 11 2023 (see internal notes)

**Pull request merged**